### PR TITLE
Add test coverage, enable checkstyle for generated code and fix offenders, enable JDK 10 GA builds.

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg [hunk 1]
```diff
@@ -51,7 +51,8 @@ public interface Mutable<name>Collection extends <name>Iterable
     default boolean removeIf(<name>Predicate predicate)
     {
         boolean changed = false;
-        for (Mutable<name>Iterator iterator = this.<type>Iterator(); iterator.hasNext();)
+        Mutable<name>Iterator iterator = this.<type>Iterator();
+        while (iterator.hasNext())
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg]: Why this change?

File: eclipse-collections-code-generator/src/main/resources/api/list/immutablePrimitiveList.stg [hunk 2]
```diff
@@ -14,17 +14,13 @@ body(type, name) ::= <<
...
-import java.lang.Iterable;
-
 import org.eclipse.collections.api.<name>Iterable;
 import org.eclipse.collections.api.block.function.primitive.<name>IntToObjectFunction;
 import org.eclipse.collections.api.block.function.primitive.<name>ToObjectFunction;
 import org.eclipse.collections.api.block.predicate.primitive.<name>Predicate;
 import org.eclipse.collections.api.block.procedure.primitive.<name>Procedure;
 import org.eclipse.collections.api.collection.primitive.Immutable<name>Collection;
 import org.eclipse.collections.api.list.ImmutableList;
-import org.eclipse.collections.api.list.ListIterable;
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/api/list/immutablePrimitiveList.stg]: Are these unused imports?

File: eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg (hunk 1 above)
Comment (nikhilnanivadekar) [reply eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg]: It is for one of the checkstyle fixes. I can revert and add a space after the ; in the precious version. But logic wise it'll do the same thing or am I missing something

File: eclipse-collections-code-generator/src/main/resources/api/list/immutablePrimitiveList.stg (hunk 2 above)
Comment (nikhilnanivadekar) [reply eclipse-collections-code-generator/src/main/resources/api/list/immutablePrimitiveList.stg]: It'll either be redundant or unused. I have segregated the commits accordingly for easier review

File: eclipse-collections-code-generator/src/main/resources/test/stack/mutable/abstractPrimitiveStackTestCase.stg [hunk 3]
```diff
@@ -297,8 +297,35 @@ public abstract class Abstract<name>StackTestCase extends Abstract<name>Iterable
...
         Verify.assertThrows(IllegalArgumentException.class, () -> iterable.chunk(0));
         Verify.assertThrows(IllegalArgumentException.class, () -> iterable.chunk(-1));
+        Verify.assertThrows(IllegalArgumentException.class, () -> this.newWith(<["0"]:(literal.(type))(); separator=", ">).chunk(-1));
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void getFirst()
+    {
+        this.classUnderTest().getFirst();
+    }
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/test/stack/mutable/abstractPrimitiveStackTestCase.stg]: Is this new test coverage?

Comment (nikhilnanivadekar) [reply eclipse-collections-code-generator/src/main/resources/test/stack/mutable/abstractPrimitiveStackTestCase.stg]: Yes

File: eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg (hunk 1 above)
Comment (donraab) [reply eclipse-collections-code-generator/src/main/resources/api/collection/mutablePrimitiveCollection.stg]: ok

**Comment by motlin** [comment]
wow 👍
